Calibration/HcalCalibAlgos/plugins/HcalIsoTrackAnalyzer.cc

```diff
@@ -121,9 +121,9 @@ void HcalIsoTrackAnalyzer::analyze(edm::Event const& iEvent, edm::EventSetup con
     if (debug)
       edm::LogVerbatim("HcalIsoTrack") << "Finds HcalIsoTrkCalibVariablesCollection with " << isotrkCalib->size()
                                        << " entries";
-    int k(0);
 #endif
-    for (auto itr = isotrkCalib->begin(); itr != isotrkCalib->end(); ++itr) {
+    int k(0);
+    for (auto itr = isotrkCalib->begin(); itr != isotrkCalib->end(); ++itr, ++k) {
```

Comment:
This could be a range based loop, right? 
```suggestion
    for (auto isotrkCalib : isotrkCalibColl) {
```

And then you'll also not need the line
` auto isotrkCalib = isotrkCalibColl.product();` from earlier

Reply:
Maybe this could be done more systematically in all the HCAL for loops, let me know if you prefer to do that, or fix the cases in the `HcalIsoTrackFilter` alone

Reply:
I shall do this later systematically in other loops. Let this PR to be merged now